From 0af99d8d5ba35118e8799684a2d4a8ea4b0c6957 Mon Sep 17 00:00:00 2001
From: yokogawa-k <yokogawa-k@klab.com>
Date: Mon, 26 Mar 2018 17:11:29 +0900
Subject: [PATCH] fix non-versioned lua not search in cmake

"FOREACH" ignore empty elements, but "FOREACH IN LISTS" include emoty elements.
https://cmake.org/cmake/help/latest/command/foreach.html
---
 cmake/FindLua.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindLua.cmake b/cmake/FindLua.cmake
index 1f95bf9..6d33dbe 100644
--- a/cmake/FindLua.cmake
+++ b/cmake/FindLua.cmake
@@ -39,7 +39,7 @@ SET(_POSSIBLE_LUA_INCLUDE include include/lua)
 SET(_POSSIBLE_SUFFIXES "52" "5.2" "-5.2" "53" "5.3" "-5.3" "")
 
 # Set up possible search names and locations
-FOREACH(_SUFFIX ${_POSSIBLE_SUFFIXES})
+FOREACH(_SUFFIX IN LISTS _POSSIBLE_SUFFIXES)
   LIST(APPEND _POSSIBLE_LUA_INCLUDE "include/lua${_SUFFIX}")
   LIST(APPEND _POSSIBLE_LUA_EXECUTABLE "lua${_SUFFIX}")
   LIST(APPEND _POSSIBLE_LUA_COMPILER "luac${_SUFFIX}")
